Dockerfile

```diff
@@ -1,41 +1,51 @@
-FROM osgeo/gdal:ubuntu-small-3.4.1 as base
+FROM osgeo/gdal:ubuntu-small-3.4.1 as base-all
 LABEL maintainer "info@camptocamp.org"
 
 # Workaround for setuptools >= 60.0.0
 ENV SETUPTOOLS_USE_DISTUTILS=stdlib
 
-RUN \
-  apt update && \
-  DEBIAN_FRONTEND=noninteractive apt install --assume-yes --no-install-recommends \
-  libmapnik3.0 \
-  mapnik-utils \
+RUN --mount=type=cache,target=/var/lib/apt/lists \
+    --mount=type=cache,target=/var/cache,sharing=locked \
+  apt-get update && \
+  DEBIAN_FRONTEND=noninteractive apt-get install --assume-yes --no-install-recommends \
+  libmapnik3.0 mapnik-utils \
   libdb5.3 \
   fonts-dejavu \
   node-carto \
-  optipng \
-  jpegoptim \
+  optipng jpegoptim \
   postgresql-client-12 net-tools iputils-ping \
-  python3-pip && \
-  apt clean && \
-  rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/* /root/.cache/*
+  python3-pip python3-wheel
 
-COPY requirements.txt Pipfile Pipfile.lock /app/
-RUN \
+
+FROM base-all as poetry
+
+WORKDIR /tmp
+COPY requirements.txt ./
+RUN --mount=type=cache,target=/root/.cache \
+    python3 -m pip install --disable-pip-version-check --requirement=requirements.txt && \
+    rm requirements.txt
+
+COPY poetry.lock pyproject.toml ./
+RUN poetry export --output=requirements.txt && \
+    poetry export --dev --output=requirements-dev.txt
```

Comment:
I do not sure understand, we install requirements.txt, after this we install with poetry and export a new requirements.